identity-account/src/account/account.rs

```diff
@@ -418,6 +428,52 @@ impl Account {
     Ok(())
   }
 
+  /// Push all unpublished changes for the given identity to the tangle in a single message.
+  pub async fn publish_changes<K: IdentityKey>(&self, key: K) -> Result<()> {
+    let identity: IdentityId = self.try_resolve_id(key).await?;
+
+    // Get the last commit generation that was published to the tangle.
+    let last_published = self.store.published_generation(identity).await?.unwrap_or_default();
+
+    // Get the commits that need to be published.
+    let commits = self.store.collect(identity, last_published).await?;
+
+    if commits.is_empty() {
+      return Ok(());
+    }
+
+    // Load the snapshot that represents the state on the tangle.
+    let snapshot = self.load_snapshot_at(identity, last_published).await?;
+
+    self.publish(snapshot, commits, true).await?;
+
+    Ok(())
+  }
+
+  /// Publishes according to the autopublish configuration.
+  async fn publish(&self, snapshot: IdentitySnapshot, commits: Vec<Commit>, force: bool) -> Result<()> {
+    if force || self.config.autopublish {
+      let id = snapshot.id();
+
+      match Publish::new(&commits) {
+        Publish::Auth => self.process_auth_change(snapshot).await?,
+        Publish::Diff => self.process_diff_change(snapshot).await?,
+        Publish::None => {}
+      }
+
+      if !commits.is_empty() {
+        let last_commit_generation = commits.last().unwrap().sequence();
+        // Publishing adds an AuthMessage or DiffMessage event, that contains the message id
+        // which is required to be set for subsequent updates.
+        // The next snapshot that loads the tangle state will require this message id to be set.
+        let generation = Generation::from_u32(last_commit_generation.to_u32() + 1);
+        self.store.set_published_generation(id, generation).await?;
+      }
+    }
```

Comment:
I think there's a potential race condition with `set_published_generation` here but also updates and `publish_changes` in general? The potential effects of a race condition don't seem to be fatal, e.g. double publishing a commit, but we should discuss it, unless I'm wrong? 

Fixing it would probably require fine-grained `RwLocks` on each of the local `IdentityId` entries for most operations.